src/ca/wolfram/beta/symath/MathNode.java

```diff
@@ -0,0 +1,25 @@
+package ca.wolfram.beta.symath;
+
+import java.util.List;
+
+public interface MathNode {
+
+    // static MathNode create(ArrayList<MathNode> nodes);
+    // static MathNode create(MathNode ... nodes);
+    NodeType getType();
+
+    List<MathNode> getChildren();
+
+    boolean isConstant();
+
+    double eval();
+
+    /**
+     * Simplifies node & its children
+     *
+     * @return true if result is a constant
```

Comment:
say something went wrong with the process. like we couldn't figure out the best way to simplfy or something.